src/System.IO/tests/CallTrackingStream.cs

```diff
@@ -0,0 +1,210 @@
+            FlushAsyncCancellationToken = cancellationToken;
+            return Inner.FlushAsync(cancellationToken);
+        }
+
+        public override int Read(byte[] buffer, int offset, int count)
+        {
+            UpdateCallCount();
+            ReadBuffer = buffer;
+            ReadOffset = offset;
+            ReadCount = count;
+            return Inner.Read(buffer, offset, count);
+        }
+
+        public override Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        {
+            UpdateCallCount();
+            ReadAsyncBuffer = buffer;
+            ReadAsyncOffset = offset;
+            ReadAsyncCount = count;
+            ReadAsyncCancellationToken = cancellationToken;
+            return Inner.ReadAsync(buffer, offset, count, cancellationToken);
+        }
+
+        public override int ReadByte()
+        {
+            UpdateCallCount();
+            return Inner.ReadByte();
+        }
+
+        public override long Seek(long offset, SeekOrigin origin)
+        {
+            UpdateCallCount();
+            SeekOffset = offset;
+            SeekOrigin = origin;
+            return Inner.Seek(offset, origin);
+        }
+
+        public override void SetLength(long value)
+        {
+            UpdateCallCount();
+            SetLengthValue = value;
+            Inner.SetLength(value);
+        }
+
+        public override void Write(byte[] buffer, int offset, int count)
+        {
+            UpdateCallCount();
+            WriteBuffer = buffer;
+            WriteOffset = offset;
+            WriteCount = count;
+            Inner.Write(buffer, offset, count);
+        }
+
+        public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        {
+            UpdateCallCount();
+            WriteAsyncBuffer = buffer;
+            WriteAsyncOffset = offset;
+            WriteAsyncCount = count;
+            WriteAsyncCancellationToken = cancellationToken;
+            return Inner.WriteAsync(buffer, offset, count, cancellationToken);
+        }
+
+        public override void WriteByte(byte value)
+        {
+            UpdateCallCount();
+            WriteByteValue = value;
+            Inner.WriteByte(value);
+        }
+
+        // Bookkeeping logic
+
+        public int TimesCalled(string member) => _callCounts.GetOrDefault(member);
+
+        private T Read<T>(T property, [CallerMemberName] string member = null)
+        {
+            IncrementOrAddCallCount(member);
+            return property;
+        }
+
+        private void Update(Action setter, [CallerMemberName] string member = null)
+        {
+            IncrementOrAddCallCount(member);
+            setter();
+        }
+
+        private void UpdateCallCount([CallerMemberName] string member = null)
+        {
+            IncrementOrAddCallCount(member);
+        }
+
+        private void IncrementOrAddCallCount(string member)
```

Comment:
Is this function necessary?  Seems it could just be combined into UpdateCallCount, and then all of the call sites for it would just call UpdateCallCount instead.